### PR TITLE
Add initBuckets functionality

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

File: turtle.go [hunk 1]
```diff
@@ -84,6 +88,23 @@ type Turtle struct {
 	closed uint32
 }
 
+func (t *Turtle) initBuckets() (err error) {
+	t.mux.Lock()
```
Comment (OneOfOne) [inline turtle.go]: Not needed since this is only called from New

**Review by OneOfOne — CHANGES_REQUESTED** [review]
Small nitpick

File: turtle.go (hunk 1 above)
Comment (itsmontoya) [reply turtle.go]: Fair enough!

**Review by OneOfOne — APPROVED** [review]

**Review by shahzilabid — APPROVED** [review]
Great idea..love it

**Comment by itsmontoya** [comment]
Woohoo! Thanks guys! :100:

**Pull request merged**